From a0ecb3794f00e65cd2a073b2073e32d6c2fb7180 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Fri, 27 Jun 2014 09:45:53 +0200
Subject: [PATCH] Fix Bug #67530 	error_log=syslog ignored

As for Bug #64915
Systemd use --nodaemonize but need to error_log to file or syslog to be honoured
---
 sapi/fpm/fpm/fpm_stdio.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sapi/fpm/fpm/fpm_stdio.c b/sapi/fpm/fpm/fpm_stdio.c
index d81e101..769d3a6 100644
--- a/sapi/fpm/fpm/fpm_stdio.c
+++ b/sapi/fpm/fpm/fpm_stdio.c
@@ -268,7 +268,11 @@ int fpm_stdio_open_error_log(int reopen) /* {{{ */
 	if (!strcasecmp(fpm_global_config.error_log, "syslog")) {
 		openlog(fpm_global_config.syslog_ident, LOG_PID | LOG_CONS, fpm_global_config.syslog_facility);
 		fpm_globals.error_log_fd = ZLOG_SYSLOG;
+#if HAVE_UNISTD_H
+		if (fpm_global_config.daemonize || (!isatty(STDERR_FILENO) && !fpm_globals.force_stderr)) {
+#else
 		if (fpm_global_config.daemonize) {
+#endif
 			zlog_set_fd(fpm_globals.error_log_fd);
 		}
 		return 0;
-- 
1.9.2

